mlrun/api/launcher.py

```diff
@@ -168,16 +168,29 @@ def enrich_runtime(
         self,
         runtime: "mlrun.runtimes.base.BaseRuntime",
         project_name: Optional[str] = "",
+        full: bool = True,
     ):
         """
         Enrich the runtime object with the project spec and metadata.
         This is done only on the server side, since it's the source of truth for the project, and we want to keep the
         client side enrichment as minimal as possible.
+        :param runtime:         the runtime object to enrich
+        :param project_name:    the project name of the project to enrich the runtime with
+        :param full:            whether to enrich the runtime with the project's full spec (before run)
+                                e.g. mount, service account, etc.
         """
-        if self._auth_info:
-            mlrun.api.api.utils.apply_enrichment_and_validation_on_function(
-                runtime, self._auth_info
-            )
+
+        # if auth given in request ensure the function pod will have these auth env vars set, otherwise the job won't
+        # be able to communicate with the api
+        mlrun.api.api.utils.ensure_function_has_auth_set(
+            runtime, self._auth_info, allow_empty_access_key=not full
+        )
+
+        if full:
+            self._enrich_full_spec(runtime)
```

Comment:
why not resolving the full from here?

```py
full_enrichment = fn.kind in ["nuclio"]
```

Reply:
When running a function we want full enrichment regardless of the kind